### PR TITLE
Use meshgrid() function from dpctl.tensor.

### PR DESCRIPTION
**Pull request by npolina4**
Use meshgrid() function from dpctl.tensor module instead of DPNP backend implementation.
A cython code relating to meshgrid() call is cleaned up.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/test_arraycreation.py [hunk 1]
```diff
@@ -507,7 +507,21 @@ def test_dpctl_tensor_input(func, args):
...
+                         ids=['[]', '[[1]]', '[[1, 2, 3], [4, 5, 6]]', '[[1, 2], [3, 4], [5, 6]]'])
+@pytest.mark.parametrize("dtype", get_all_dtypes(no_float16=False))
+@pytest.mark.parametrize("indexing",
+                         ["ij", "xy"],
+                         ids=["ij", "xy"])
+def test_meshgrid(arrays, dtype, indexing):
+    func = lambda xp, xi: xp.meshgrid(*xi, indexing=indexing)
+    a = tuple(numpy.array(array, dtype=dtype) for array in arrays)
+    ia = tuple(dpnp.array(array, dtype=dtype) for array in arrays)
+    assert_array_equal(func(numpy, a), func(dpnp, ia))
```
Comment (antonwolfy) [inline tests/test_arraycreation.py:527]: new line at the end of file is missing

File: tests/test_arraycreation.py [hunk 2]
```diff
@@ -507,7 +507,21 @@ def test_dpctl_tensor_input(func, args):
...
+
+
+@pytest.mark.parametrize("arrays",
+                         [[], [[1]], [[1, 2, 3], [4, 5, 6]], [[1, 2], [3, 4], [5, 6]]],
+                         ids=['[]', '[[1]]', '[[1, 2, 3], [4, 5, 6]]', '[[1, 2], [3, 4], [5, 6]]'])
+@pytest.mark.parametrize("dtype", get_all_dtypes(no_float16=False))
+@pytest.mark.parametrize("indexing",
+                         ["ij", "xy"],
+                         ids=["ij", "xy"])
+def test_meshgrid(arrays, dtype, indexing):
```
Comment (antonwolfy) [inline tests/test_arraycreation.py:523]: Could you please add 1 more test to secure CFD and 1 extra test to check if usm_type of output array is the same as input one.

File: dpnp/dpnp_container.py [hunk 3]
```diff
@@ -180,6 +180,15 @@ def eye(N,
     return dpnp_array(array_obj.shape, buffer=array_obj, order=order)
 
 
+def meshgrid(*xi, indexing="xy"):
+    """Creates list of `dpnp_array` coordinate matrices from vectors."""
+    if len(xi) == 0:
+        return []
```
[the remark this answers is not in the document]
Comment (npolina4) [reply dpnp/dpnp_container.py:186]: It will move to dpctl.meshgrid()

**Review by antonwolfy — COMMENTED** [review]
Thank you! LGTM!

**Review by antonwolfy — APPROVED** [review]

**Pull request merged**